### PR TITLE
feat(nowait): allow waitUntil:nowait for actions

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/frames.ts [hunk 1]
```diff
@@ -100,7 +100,9 @@ export class FrameManager {
     }
   }
 
-  async waitForNavigationsCreatedBy<T>(action: () => Promise<T>, options?: types.WaitForNavigationOptions): Promise<T> {
+  async waitForNavigationsCreatedBy<T>(action: () => Promise<T>, options?: types.ActionWaitOptionsNoWaitFor): Promise<T> {
+    if (options && options.waitUntil && options.waitUntil === 'nowait')
```
Comment (dgozman) [inline src/frames.ts]: nit: no need to check `options.waitUntil` for truthy :)

File: docs/api.md [hunk 2]
```diff
@@ -2174,6 +2194,13 @@ console.log(frame === contentFrame);  // -> true
     - y <[number]>
   - `modifiers` <[Array]<"Alt"|"Control"|"Meta"|"Shift">> Modifier keys to press. Ensures that only these modifiers are pressed during the hover, and then restores current modifiers back. If not specified, currently pressed modifiers are used.
   - `waitFor` <[boolean]> Whether to wait for the element to be present in the dom and displayed (for example, no `display:none`), stop moving (for example, wait until css transition finishes) and potentially receive pointer events at the click point (for example, wait until element becomes non-obscured by other elements). Defaults to `true`.
+  - `waitUntil` <"commit"|"load"|"domcontentloaded"|"networkidle0"|"networkidle2"|"nowait"> Actions that cause navigations are waiting for those navigations to `commit` by default. This behavior can be changed to either wait for another load phase or to omit the waiting altogether using `nowait`:
```
Comment (dgozman) [inline docs/api.md]: Ditto.

File: docs/api.md [hunk 3]
```diff
@@ -1276,6 +1280,13 @@ Shortcut for [page.mainFrame().goto(url[, options])](#framegotourl-options)
...
```
Comment (dgozman) [inline docs/api.md]: Why is hover waiting for navigaiton?

**Review by dgozman — APPROVED** [review]

File: src/frames.ts [hunk 4]
```diff
@@ -1115,7 +1117,7 @@ class PendingNavigationBarrier {
 
   async addFrame(frame: Frame) {
     this.retain();
-    await frame.waitForNavigation(this._waitOptions).catch(e => {});
+    await frame.waitForNavigation(this._options as types.NavigateOptions).catch(e => {});
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply src/frames.ts]: trimming the 'nowait' here...

**Pull request merged**